src/app/components/week-selector/week-selector.component.ts

```diff
@@ -0,0 +1,69 @@
+import { async } from '@angular/core/testing';
+import { Component, OnInit, Input, Output, EventEmitter, AfterContentChecked } from '@angular/core';
+import { addWeeks, subWeeks, subYears, getWeek, getYear } from 'date-fns';
+
+@Component({
+  selector: 'app-week-selector',
+  templateUrl: './week-selector.component.html',
+  styleUrls: ['./week-selector.component.scss']
+})
+export class WeekSelectorComponent implements OnInit {
+
+  @Input() currentDate: Date;
+  @Input() startDate: Date;
+  @Input() subTitle: string;
+
+  @Output() passDate = new EventEmitter<number>();
+
+  weekNumber: number;
+  year: number;
+  isCurrentWeek: boolean;
+  isStartingWeek: boolean;
+  disabledLeft: boolean = false;
+  disabledRight: boolean;
+  date: Date
+
+  constructor() {}
+
+  ngOnInit() {
+    this.isCurrentWeek = true;
+    this.disabledRight = true;
+    this.date = this.currentDate
+    this.weekNumber = getWeek(this.date);
+    this.passDate.emit(this.weekNumber)
+    this.year = getYear(this.date)
+  }
+
+  onPreviousWeek() {
```

Comment:
Wat doet deze method?

Reply:
Deze haalt een week van het week nummer af en emit deze. Checkt ook of de deze previous week niet voor de voor de start datum uit komt.

Reply:
Deze comment heeft me getriggered er een enkele functie van te maken. Enige verschil was eigenlijk alleen plus of min een week.

Reply:
Het ging me vooral om de naamgeving 'onPreviousWeek'. :) 

Nu vind ik hem eigenlijk nog steeds niet zo duidelijk. `selectWeek(true)` zegt voor mij iets als: 'selecteer de week'.

Ik zou dan iets als `selectWeek({ next: true })` instellen, of liever nog opsplitsen in drie verschillende methods: `public selectPrevWeek(): void`, `public selectNextWeek(): void` en `private selectWeek(next: boolean): void`. Die laatste wordt dan intern aangeroepen in de twee andere methods.

Maar goed, dit is ook een project voor jou zelf alleen volgens mij, dus dan zou je ervoor kunnen gaan dat jij het begrijpt en dat dat prima is.